astropy/coordinates/baseframe.py

```diff
@@ -224,50 +236,111 @@ def __init__(self, *args, **kwargs):
+                differential_data = diffs.get('s', None)
+                if ((differential_data is None and len(diffs) > 0) or
+                        (differential_data is not None and len(diffs) > 1)):
+                    raise ValueError('Multiple differentials are associated '
+                                     'with the representation object passed in '
+                                     'to the frame initializer. Only a single '
+                                     'velocity differential is supported. Got: '
+                                     '{0}'.format(diffs))
+
         elif self.representation:
+            # Get any representation data passed in to the frame initializer
+            # using keyword or positional arguments for the component names
             repr_kwargs = {}
             for nmkw, nmrep in self.representation_component_names.items():
                 if len(args) > 0:
-                    #first gather up positional args
+                    # first gather up positional args
                     repr_kwargs[nmrep] = args.pop(0)
                 elif nmkw in kwargs:
                     repr_kwargs[nmrep] = kwargs.pop(nmkw)
 
-            #special-case the Spherical->UnitSpherical if no `distance`
-            #TODO: possibly generalize this somehow?
+            # special-case the Spherical->UnitSpherical if no `distance`
+            # TODO: possibly generalize this somehow?
 
             if repr_kwargs:
                 if repr_kwargs.get('distance', True) is None:
                     del repr_kwargs['distance']
-                if (issubclass(self.representation, SphericalRepresentation) and
-                        'distance' not in repr_kwargs):
+
+                if (issubclass(self.representation, r.SphericalRepresentation)
+                        and 'distance' not in repr_kwargs):
                     representation = self.representation._unit_representation
                 else:
                     representation = self.representation
+
                 representation_data = representation(copy=copy, **repr_kwargs)
 
+            # Now we handle the Differential data:
+            # Get any differential data passed in to the frame initializer
+            # using keyword or positional arguments for the component names
+            diff_kwargs = {}
+            for nmkw, nmrep in self.differential_component_names.items():
+                if len(args) > 0:
+                    # first gather up positional args
+                    diff_kwargs[nmrep] = args.pop(0)
+                elif nmkw in kwargs:
+                    diff_kwargs[nmrep] = kwargs.pop(nmkw)
+
+            if diff_kwargs:
+                if (hasattr(self.differential_cls, '_unit_differential') and
+                        'd_distance' not in diff_kwargs):
+                    differential = self.differential_cls._unit_differential
+
+                elif len(diff_kwargs) == 1 and 'd_distance' in diff_kwargs:
```

Comment:
Maybe overkill, but what if there are subclasses? So perhaps `and not issubclass(self.differential_cls, r.RadialDifferential)`?